default docker config: gateway application crashes with error (not able to connect elasticsearch)

##### **Overview of the issue**
in Jhipster 5.3.1 default docker configuration do not work
##### **Motivation for or Use Case**
generated simple config should work
##### **Reproduce the error**
generate new project
1. run Jhipster for microservice : application gateway
(maybe not only microservice problem)
**then start the application from generated  docker configuration in 
gateway/src/main/docker/app.yml**

docker-compose -f src/main/docker/app.yml up -d

gateway application crash with error (not able to connect elasticsearch)
from logs:
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'userSearchRepository': Invocation of init method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.data.elasticsearch.repository.support.NumberKeyedRepository]: Constructor threw exception; nested exception is org.springframework.data.elasticsearch.ElasticsearchException: failed to execute action

....
io.searchbox.client.config.exception.CouldNotConnectException: Could not connect to http://localhost:9200
...
Caused by: java.net.ConnectException: Connection refused (Connection refused)

but 
from browser http://localhost:9200   responds correctly !!!
##### **Suggest a Fix**

<!-- For bug reports, if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**

jhipster 5.3.1

##### **JHipster configuration**

Executing jhipster:info
Options: 
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
report@0.0.0 /home/ond/prog/training/java/jhipster-demo/reporting-ond-api/gateway
└── generator-jhipster@5.3.1 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "wr.gw",
      "nativeLanguage": "sk"
    },
    "jhipsterVersion": "5.3.1",
    "applicationType": "gateway",
    "baseName": "report",
    "packageName": "wr.gw",
    "packageFolder": "wr/gw",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "postgresql",
    "prodDatabaseType": "postgresql",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "gradle",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": removed
    "clientFramework": "react",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "sk",
    "languages": [
      "sk"
    ]
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

The passed validation must be valid to be added to the field.
Errors: No value
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: ERROR! 
Error while parsing entities to JDL


##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

##### **Browsers and Operating System**
debian9  linux
firefox 
<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

- [ ] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/kubernetes/templates/deployment.yml.ejs
+++ b/generators/kubernetes/templates/deployment.yml.ejs
@@ -134,7 +134,7 @@ spec:
         <%_ } _%>
         <%_ if (app.searchEngine === 'elasticsearch') { _%>
         - name: SPRING_DATA_JEST_URI
-          value: <%= app.baseName.toLowerCase() %>-elasticsearch.<%= kubernetesNamespace %>.svc.cluster.local:9300
+          value: http://<%= app.baseName.toLowerCase() %>-elasticsearch.<%= kubernetesNamespace %>.svc.cluster.local:9200
         <%_ } _%>
         <%_ if (app.messageBroker === 'kafka') { _%>
         - name: SPRING_CLOUD_STREAM_KAFKA_BINDER_BROKERS

--- a/generators/openshift/templates/deployment.yml.ejs
+++ b/generators/openshift/templates/deployment.yml.ejs
@@ -228,7 +228,7 @@ objects:
             <%_ } _%>
             <%_ if (app.searchEngine === 'elasticsearch') { _%>
             - name: SPRING_DATA_JEST_URI
-              value: ${APPLICATION_NAME}-elasticsearch:9300
+              value: http://${APPLICATION_NAME}-elasticsearch:9200
             <%_ } _%>
             <%_ if (app.messageBroker === 'kafka') { _%>
             - name: SPRING_CLOUD_STREAM_KAFKA_BINDER_BROKERS

--- a/generators/server/templates/src/main/docker/app.yml.ejs
+++ b/generators/server/templates/src/main/docker/app.yml.ejs
@@ -86,7 +86,7 @@ services:
                 <%_ } _%>
             <%_ } _%>
             <%_ if (searchEngine === 'elasticsearch') { _%>
-            - SPRING_DATA_JEST_URI=<%= baseName.toLowerCase() %>-elasticsearch:9300
+            - SPRING_DATA_JEST_URI=http://<%= baseName.toLowerCase() %>-elasticsearch:9200
             <%_ } _%>
             <%_ if (messageBroker === 'kafka') { _%>
             - SPRING_CLOUD_STREAM_KAFKA_BINDER_BROKERS=kafka

--- a/test/templates/compose/03-psql/src/main/docker/app.yml
+++ b/test/templates/compose/03-psql/src/main/docker/app.yml
@@ -6,7 +6,7 @@ services:
             - SPRING_PROFILES_ACTIVE=prod
             - SPRING_CLOUD_CONFIG_URI=http://admin:admin@jhipster-registry:8761/config
             - SPRING_DATASOURCE_URL=jdbc:postgresql://mspsql-postgresql:5432/mspsql
-            - SPRING_DATA_JEST_URI=mspsql-elasticsearch:9300
+            - SPRING_DATA_JEST_URI=http://mspsql-elasticsearch:9200
     mspsql-postgresql:
         extends:
             file: postgresql.yml

